src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int MAX_VALUE = 50;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        switch (figureNumber) {
+            case 0:
+                int radius = random.nextInt(MAX_VALUE);
+                Circle circle = new Circle(radius);
+                circle.setColor(colorSupplier.getRandomColor());
+                return circle;
+            case 1:
+                int at = random.nextInt(MAX_VALUE);
+                int bt = random.nextInt(MAX_VALUE);
+                int ht = random.nextInt(MAX_VALUE);
+                IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid(at, bt, ht);
+                isoscelesTrapezoid.setColor(colorSupplier.getRandomColor());
+                return isoscelesTrapezoid;
+            case 2:
```

Comment:
```suggestion
                return getRandomCircle();
            case 1:
                return getRandomIsoscelesTrapezoid();
            case 2:
```
lets divide the logic into smaller private methods - it makes it easier to read your code